doc/admin-guide/plugins/filter_body.en.rst

```diff
@@ -0,0 +1,440 @@
+- Per-rule metrics counters for monitoring match activity.
+
+Installation
+============
+
+The ``filter_body`` plugin is an experimental plugin. To build it, either pass
+``-DENABLE_FILTER_BODY=ON`` to ``cmake`` when configuring the build::
+
+    cmake -DENABLE_FILTER_BODY=ON ...
+
+Or enable all experimental plugins with ``-DBUILD_EXPERIMENTAL_PLUGINS=ON``::
+
+    cmake -DBUILD_EXPERIMENTAL_PLUGINS=ON ...
+
+Configuration
+=============
+
+The plugin is configured as a remap plugin with a YAML configuration file::
+
+    map http://example.com/ http://origin.example.com/ @plugin=filter_body.so @pparam=filter_body.yaml
+
+The configuration file path can be relative to the |TS| configuration directory
+or an absolute path.
+
+Configuration File Format
+-------------------------
+
+The configuration file uses YAML format with a list of rules. Each rule has a
+``name``, a ``filter`` section containing all filtering criteria, and an
+``action`` section specifying what to do when a match occurs::
+
+    rules:
+      - name: rule_name
+        filter:
+          direction: request    # optional, defaults to request
+          methods:              # for request rules only
+            - POST
+            - PUT
+          max_content_length: 1048576
+          headers:
+            - name: Content-Type
+              patterns:
+                - "application/xml"
+                - "text/xml"
+          body_patterns:
+            - "<!ENTITY"
+            - "<!DOCTYPE"
+        action:
+          - log
+          - block
+          - add_header:
+              X-Security-Match: <rule_name>
+              X-Another-Header: some-value
+
+For response rules, use ``status`` instead of ``methods`` within the ``filter``
+section::
+
+    rules:
+      - name: response_rule
+        filter:
+          direction: response
+          status:               # for response rules only
+            - 200
+            - 201
+          body_patterns:
+            - "sensitive_data"
+        action:
+          - log
+
+Rule Options
+------------
+
+``name`` (required)
+    A unique name for the rule. Used in log messages and metrics when the rule
+    matches. The special placeholder ``<rule_name>`` can be used in header values
+    to substitute the rule's name dynamically.
+
+``filter`` (required)
+    A section containing all filtering criteria that determine which requests or
+    responses the rule applies to. This section separates the "what to match"
+    from the "what to do" (action).
+
+Filter Options
+--------------
+
+The following options are valid within the ``filter`` section:
+
+``direction`` (optional)
+    Specifies whether to inspect request or response bodies.
+    Valid values: ``request``, ``response``. Default: ``request``.
+
+``methods`` (optional)
+    List of HTTP methods to match. If not specified, all methods are matched.
+    Only valid for request rules. Example: ``[GET, POST, PUT]``.
+
+``status`` (optional)
```

Comment:
I would reorder this and have subsections of options that are only valid for request rules and ones that are only valid for response rules.